Cluster for stage analytics-jenkins.

dcadams opened the pull request.

dcadams commented:
Corresponding secure pr:
https://github.com/Stanford-Online/configuration-secure/pull/325

dcadams commented:
@stvstnfrd mind giving this an eyeball?

stvstnfrd commented on stanford/playbooks/analytics-jenkins.yml:
```diff
@@ -0,0 +1,23 @@
+  gather_facts: True
+  become: True
+  vars:
+    CLUSTER_NAME: analytics-jenkins
```
We may want to consider choosing an alternative name for this.

Two issues I ran into with naming were:
- length restrictions
- character restrictions

One potential hiccup I ran into with other clusters was that the names of some resources (ELBs, security groups, etc.) had surprisingly short maximum lengths for naming. I can't remember which resources, but I definitely remember it happening with `elasticsearch-forum`. Some resources were fine, but then you'd hit some examples like `lagunita-group-elasticsearch-forum-elb-public` which would fail.
Again, I don't remember what all was affected, but it's something to consider.

The other issue was with special characters. It may have been hyphens and/or underscores (again, I admittedly don't remember, sorry), but that prevented some resources from being created.

Given that, I've been attempting to keep cluster names short and strictly using lowercase alpha characters.

So, not an absolute deal breaker, but if we can find an alternative, great. If not, it'll be something to keep an out for.

The first thing that jumps to mind is "scheduler", though maybe that's too ambiguous?
I'll brainstorm over the weekend.

stvstnfrd commented on stanford/roles/cluster/defaults/main.yml:
```diff
@@ -90,6 +95,19 @@ CLUSTER_SUBNET: "{{ CLUSTER_SUBNETS[CLUSTER_NAME][deployment_subnet_index|int] }
+    - proto: tcp
+      from_port: 80
+      to_port: 80
+      cidr_ip: '0.0.0.0/0'
```
You can actually lock this down to `"{{ DEPLOYMENT_CIDR }}"`,
since the ELB is located inside the VPC.

stvstnfrd commented on stanford/roles/cluster/defaults/main.yml:
```diff
@@ -90,6 +95,19 @@ CLUSTER_SUBNET: "{{ CLUSTER_SUBNETS[CLUSTER_NAME][deployment_subnet_index|int] }
+      cidr_ip: '0.0.0.0/0'
+    - proto: tcp
+      from_port: 8080
+      to_port: 8080
```
You should be able to drop this rule.
Nginx runs on the jenkins boxes and automatically routes port `80` externally -> `8080` locally.

stvstnfrd commented on stanford/roles/cluster/defaults/main.yml:
```diff
@@ -188,6 +206,32 @@ CLUSTER_SECURITY_GROUP_RULES:
+        - protocol: https
+          load_balancer_port: 443
+          instance_protocol: http
+          instance_port: 8080
```
Per note above, you can route this directly to port 80.

stvstnfrd commented on stanford/roles/cluster/defaults/main.yml:
```diff
@@ -188,6 +206,32 @@ CLUSTER_SECURITY_GROUP_RULES:
+    - name: "{{ COMMON_DEPLOYMENT }}-elb-analytics-jenkins-public"
+      health_check:
+        ping_protocol: http
+        ping_path: '/health/'
```
Are we using this on CME?
I didn't realize this endpoint existed...

dcadams replied:
yes.

stvstnfrd replied:
Per f2f, change to `cc.xml`; see `build` cluster as reference.

dcadams commented:
Addressed comments apart from the name length. Since I can't think of a better name I'm inclined to try it and then think a bit more if it causes a problem.

stvstnfrd commented on stanford/stage/analytics-jenkins1.yml:
```diff
@@ -0,0 +1,7 @@
+---
+- include: ../playbooks/analytics-jenkins.yml
+  vars:
+    CLUSTER_INSTANCE_TYPE: t2.medium
```
Since this is stage, we can drop this line and fall back to the default, `.t2.micro`, I think?
